crates/bevy_ecs/src/world/entity_fetch.rs

```diff
@@ -3,12 +3,98 @@ use core::mem::MaybeUninit;
 
 use crate::{
     entity::{hash_map::EntityHashMap, hash_set::EntityHashSet, Entity, EntityDoesNotExistError},
+    error::Result,
     world::{
         error::EntityMutableFetchError, unsafe_world_cell::UnsafeWorldCell, EntityMut, EntityRef,
         EntityWorldMut,
     },
 };
 
+/// Provides a safe interface for non-structural access to the entities in a [`World`].
```

Comment:
Reading the docs for [DeferredWorld](https://docs.rs/bevy/latest/bevy/ecs/world/struct.DeferredWorld.html), it seems like it only allows non-structural changes.  It also seems like EntityFetcher also allows only non structural changes. So wouldn't allowing this for just the world be unneeded?

Reply:
Disallowing structural changes is a matter of upholding Rust's safety guarantees. Handing out `EntityWorldMut` would break rust's mutable aliasing requirements because we would have 2 `&mut World` references.

Reply:
I think what you are saying is that EntityFetcher needs to disallow mutable changes because otherwise the returned Commands would cause safety conflicts because one could theoretically use both the EntityFetcher and the Commands to alter the world, thereby causing memory issues. 

So using commands in a world instead of a deferred world would allow structural changes, but only with the commands. Is that right?

Reply:
No structural changes are allowed for the duration of the returned `EntityFetcher` and `Commands` pair for either `World` or `DeferredWorld`; commands are only applied the next time `world.flush()` is called.

Reply:
If structural changes are allowed for a deferred world with (`EntityFetcher`, `Commands`) , then does that break deferred world? Or are the deferred world docs wrong? 

Thanks for responding btw! I appreciate it

`
/// A [`World`] reference that disallows structural ECS changes.
/// This includes initializing resources, registering components or spawning entities.
pub struct DeferredWorld<'w> {
    // SAFETY: Implementors must not use this reference to make structural changes
    world: UnsafeWorldCell<'w>,
}
`

Reply:
No, because commands are not applied until after the `DeferredWorld` goes out of scope (`DeferredWorld` is just a wrapper around a `&mut World`). You can queue commands all you want which perform structural changes, they just don't get applied until later.

Reply:
Ahhh okay, so then it sounds like with a `World` you can do something like `e = world.spawn()` whereas with a `DeferredWorld` you would need to do something like `commands.spawn()` because you can't just mutate state in the `DeferredWorld`  directly, you have to go through the command queue.